packages/caliper-core/test/manager/monitors/monitor-docker.js

```diff
@@ -0,0 +1,243 @@
+* limitations under the License.
+*/
+
+'use strict';
+
+
+const rewire = require('rewire');
+const chai = require('chai');
+const should = chai.should();
+const sinon = require('sinon');
+const DockerMonitorRewire = rewire('../../../lib/manager/monitors/monitor-docker');
+const ConfigUtil = require('../../../lib/common/config/config-util.js');
+
+describe('A Docker monitor', () => {
+
+    //stub docker stub
+    const dockerStats = {
+        id: "118",
+        mem_usage: 10000,
+        mem_limit: 10000,
+        mem_percent: 10000,
+        cpu_percent: 10000,
+        netIO: {
+          rx: 10000,
+          wx: 10000,
+        },
+        blockIO: {
+          r: 10000,
+          w: 10000,
+        },
+        cpu_stats: {
+            "cpu_usage":{"total_usage":1000000,"usage_in_kernelmode":10000,"usage_in_usermode":10000},
+            "system_cpu_usage":100000,
+            "online_cpus":8,
+            "throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}
+        },
+        precpu_stats: {
+            "cpu_usage":{"total_usage":0,"usage_in_kernelmode":0,"usage_in_usermode":0},
+            "throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0},
+            "system_cpu_usage":0
+        },
+        memory_stats: {
+            "usage":46100480,
+            "stats":{"cache": 500,"active_anon":0,"active_file":0,"anon":32440320,"anon_thp":0,"file":11624448,"file_dirty":0,"file_mapped":10407936,"file_writeback":0,"inactive_anon":29007872,"inactive_file":15003648,"kernel_stack":147456,"pgactivate":0,"pgdeactivate":0,"pgfault":10065,"pglazyfree":759,"pglazyfreed":0,"pgmajfault":66,"pgrefill":0,"pgscan":0,"pgsteal":0,"shmem":0,"slab":0,"slab_reclaimable":0,"slab_unreclaimable":0,"sock":0,"thp_collapse_alloc":0,"thp_fault_alloc":0,"unevictable":0,"workingset_activate":0,"workingset_nodereclaim":0,"workingset_refault":0},
+            "limit":2081312768
+        },
+        networks: {
+            "eth0":{"rx_bytes":10000,"rx_packets":10000,"rx_errors":0,"rx_dropped":0,"tx_bytes":10000,"tx_packets":10000,"tx_errors":0,"tx_dropped":0}
+        },
+    };
+
+
+    //create os stub
+    const osStub = {
+        cpus : sinon.stub()
+    }
+    const returnValueCpus = [ 
+    { 
+        model: 'Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz',
+        speed: 2712,
+        times: { user: 900000, nice: 0, sys: 940265, idle: 11928546, irq: 147046 } 
+    },
+    { 
+        model: 'Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz',
+        speed: 2712,
+        times: { user: 860875, nice: 0, sys: 507093, idle: 12400500, irq: 27062 } 
+    }]
+    
+    osStub.cpus.returns(returnValueCpus);
+    const revertOS = DockerMonitorRewire.__set__("os", osStub)
+
+
+    //create System information stub
+    const SystemInformationStub = {
+        dockerContainers : sinon.stub(),
+        dockerContainerStats : sinon.stub()
+    };
+    SystemInformationStub.dockerContainers.returns([{
```

Comment:
this might need to be `resolves` rather than `returns`

Reply:
yup, you are right thanks